fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBOrQueryToUnionTest.java

```diff
@@ -943,48 +960,52 @@ void testUnorderableOrQueryWithAnd(boolean removesDuplicates) throws Exception {
                         Query.field("str_value_indexed").equalsValue("even"),
                         Query.or(
                                 Query.field("num_value_3_indexed").equalsValue(1),
                                 Query.field("num_value_3_indexed").greaterThan(3))))
                 .build();
+        orQueryParams.setPlannerConfiguration(this);
 
         // Index(str_value_3_index [[even, 1],[even, 1]]) ∪[Field { 'num_value_3_indexed' None}, Field { 'num_value_unique' None}] Index(str_value_3_index ([even, 3],[even]])
         RecordQueryPlan plan = planner.plan(query);
+        final KeyExpression comparisonKey = (orQueryParams.shouldOmitPrimaryKeyInOrderingKey() || planner instanceof CascadesPlanner)
+                ? concatenateFields("num_value_3_indexed", "num_value_unique")
+                : concatenateFields("str_value_indexed", "num_value_3_indexed", "num_value_unique");
+        final BindingMatcher<? extends RecordQueryPlan> planMatcher = queryPlanMatcher(orQueryParams, List.of(
+                indexPlan().where(indexName("str_value_3_index")).and(scanComparisons(range("[[even, 1],[even, 1]]"))),
+                indexPlan().where(indexName("str_value_3_index")).and(scanComparisons(range("([even, 3],[even]]")))
+        ), comparisonKey);
+        assertMatchesExactly(plan, planMatcher);
 
         if (planner instanceof RecordQueryPlanner) {
-            final BindingMatcher<? extends RecordQueryPlan> planMatcher =
-                    fetchFromPartialRecordPlan(
-                            RecordQueryPlanMatchers.unionOnExpressionPlan(
-                                            coveringIndexPlan()
-                                                    .where(indexPlanOf(indexPlan().where(indexName("str_value_3_index")).and(scanComparisons(range("[[even, 1],[even, 1]]"))))),
-                                            coveringIndexPlan()
-                                                    .where(indexPlanOf(indexPlan().where(indexName("str_value_3_index")).and(scanComparisons(range("([even, 3],[even]]"))))))
-                                    .where(comparisonKey(concat(field("num_value_3_indexed"), field("num_value_unique")))));
-            assertMatchesExactly(plan, planMatcher);
-
-            assertEquals(-60058062, plan.planHash(PlanHashable.CURRENT_LEGACY));
-            assertEquals(-1391842890, plan.planHash(PlanHashable.CURRENT_FOR_CONTINUATION));
+            if (orQueryParams.shouldDeferFetch()) {
+                if (orQueryParams.shouldOmitPrimaryKeyInOrderingKey()) {
+                    assertEquals(-60058062, plan.planHash(PlanHashable.CURRENT_LEGACY));
+                    assertEquals(-1391842890, plan.planHash(PlanHashable.CURRENT_FOR_CONTINUATION));
+                } else {
+                    assertEquals(-664830657, plan.planHash(PlanHashable.CURRENT_LEGACY));
+                    assertEquals(1572009327, plan.planHash(PlanHashable.CURRENT_FOR_CONTINUATION));
```

Comment:
This is one of the few queries that had a changed plan hash. These plan hashes match the pre-#2350 plan hashes